service/src/main/java/bio/terra/catalog/api/PublicApiController.java

```diff
@@ -4,22 +4,47 @@
 import bio.terra.catalog.model.SystemStatus;
 import bio.terra.catalog.model.VersionProperties;
 import bio.terra.catalog.service.CatalogStatusService;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
 
 @Controller
 public class PublicApiController implements PublicApi {
   private final CatalogStatusService statusService;
   private final VersionConfiguration versionConfiguration;
 
+  private final String swaggerClientId;
+
   @Autowired
   public PublicApiController(
       CatalogStatusService statusService, VersionConfiguration versionConfiguration) {
     this.statusService = statusService;
     this.versionConfiguration = versionConfiguration;
+
+    String clientId = "";
+
+    try {
+      try (var reader =
+          new BufferedReader(
+              new InputStreamReader(
+                  new ClassPathResource("rendered/swagger-client-id").getInputStream(),
```

Comment:
For what it's worth, you don't need to use the Spring API `ClassPathResource()` here. You can instead use the Java API `getResourceAsStream()`: `getClass().getResourceAsStream("/rendered/swagger-client-id")`

This also lets us render the value into the `build` directory instead of the source directory. Although doing it that way means that if a user happens to clean their build tree they'll need to re-render the config, which is probably the reason why ECM has it set to work this way.

Reply:
code was suggested  to change it to 
`try { clientId =
          new String(
              new ClassPathResource("rendered/swagger-client-id").getInputStream().readAllBytes(),
              StandardCharsets.UTF_8);
    } `
should I change it to your way?